### ISSUE TITLE
Cannot open Pico web sites anymore after upgrade to NC 19.0.4

### ISSUE THREAD

**Issue by sm8ps** [nextcloud/cms_pico#149]
After upgrading to NC 19.0.4 from 19.0.3 I cannot open any of the Pico web sites anymore. It took me a while to even find out where things are going wrong because there had been problems with the NC app store during the upgrade. At the time being, all apps seem to be updated (with the single exception of the Community Document Server which refuses to upgrade from 0.1.7 to 0.1.8). 

I had been absent for ten days and all Pico sites were running fine before that. I do not exactly what all I changed because I was not aware of any problems. As a last resort, I tried `occ maintenance:repair` to no avail.

This is what happens on NC 19.0.4 with Pico CMS 1.0.10 when I create a new web site from an administrator account: The site data is created in the files but when I try to access the web site via HTTP, a process php-fpm starts running on one CPU core at 100% and only terminates after a long time with status code 500 (internal server error).

The NC log shows the following:
```
"app":"PHP","method":"GET","url":"/index.php/apps/cms_pico/pico/test-a","message":"Allowed memory size of 1073741824 bytes exhausted (tried to allocate 20480 bytes) at /var/www/nextcloud/lib/private/Log/ExceptionSerializer.php#137","userAgent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:80.0) Gecko/20100101 Firefox/80.0","version":"19.0.4.2"}
```

`top` confirms that the virtual memory of the process does indeed increase continuously even after the request has been canceled from the client side.

Am I the only one facing this issue? Any ideas for further trouble-shooting? Thanks in advance!

Edit: I should point out #145 which may be related.

**Comment by PhrozenByte** [comment]
[`ExceptionSerializer::removeValuesFromArgs()`](https://github.com/nextcloud/server/blob/55b923e1fdc76df77084a45b3a85d820f807936d/lib/private/Log/ExceptionSerializer.php#L136-L145) is causing an infinite loop eating up all your memory. Please open a new issue in `nextcloud/server` to fix this infinite loop first and link this issue as reference.

After the infinite loop has been fixed we can actually check what the issue is. As a temporary solution you could replace the method in your Nextcloud install with the following:

```php
	private function removeValuesFromArgs($args, $values) {
		return $args;
	}
```

Then check your `nextcloud.log`, it should now contain a proper exception, allowing us to debug the issue.

**Comment by sm8ps** [comment]
Thanks a lot for the guidance, @PhrozenByte! I have opened  nextcloud/server#23429.

In the meantime, I could upgrade *all* apps via `occ app:update --all` and have run `occ maintenance:repair` and re-loaded all themes and plugins for cms_pico just to be on the safe side.

I have tried short-cutting `removeValuesFromArgs()` as you had suggested with the result that the web server returns HTTP-502 (bad gateway) *immediately*. However, there is *nothing* appearing in the Nextcloud log. Here is the web server log:
```
192.168.1.102 - - [14/Oct/2020:08:16:46 +0200] "GET /index.php/apps/cms_pico/pico/test-a HTTP/1.0" 502 150 "https://nextcloud.wildi.name/settings/user/cms_pico" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:81.0) Gecko/20100101 Firefox/81.0" "192.168.1.166"
```

**Comment by PhrozenByte** [comment]
> I have tried short-cutting `removeValuesFromArgs()` as you had suggested with the result that the web server returns HTTP-502 (bad gateway) _immediately_. However, there is _nothing_ appearing in the Nextcloud log.

Okay, that is indeed super weird :confused: Is there something in PHP's `error.log`? Also check your webserver logs.

**Comment by sm8ps** [comment]
Some good news although no further insight: I removed the cms_pico app, removed the folder 'appdata_###/cms_pico' and re-added the app. After that I could finally create Pico-sites as usual.

It is almost impossible to say what caused the problem. No log showed anything substantial at all. I did fiddle around with the themes quite a bit and may have introduced some intricate malfunction from there. The issue with the NC upgrade not properly working for all apps misguided me into believing that it could be the cause.

Closing for now. I will re-add the modified themes and if anything shows up, I shall report back here. Thanks for the help and sorry for the noise!
